modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/components/TimelineDropdownMenu.js

```diff
@@ -3,110 +3,153 @@
 			symbolLeft: 'list-ul',
 		});
 	}
 
-	dropdownItems.push({
-		href: reviewURL,
-		label: Liferay.Language.get('review-changes'),
-		symbolLeft: 'list-ul',
-	});
+	if (
+		(timelineItem.status.code === WORKFLOW_STATUS_DRAFT ||
+			timelineItem.status.code === WORKFLOW_STATUS_EXPIRED) &&
+		!!timelineItem.actions.update &&
+		moveURL
+	) {
+		dropdownItems.push({
+			href: moveURL,
+			label: Liferay.Language.get('move-changes'),
+			symbolLeft: 'move-folder',
+		});
+	}
 
-	if (deleteURL) {
-		dropdownItems.push(
-			{type: 'divider'},
-			{
-				label: Liferay.Language.get('delete'),
-				onClick: () => {
-					Liferay.Util.openConfirmModal({
-						message: Liferay.Language.get(
-							'are-you-sure-you-want-to-delete-this-publication'
-						),
-						onConfirm: (isConfirmed) => {
-							if (isConfirmed) {
-								fetch(deleteURL, {
-									method: 'DELETE',
-								}).then((response) => {
-									if (response.ok) {
-										showNotification(
-											sub(
-												Liferay.Language.get(
-													'x-was-deleted-successfully'
-												),
-												Liferay.Language.get(
-													'publication'
-												)
-											),
-											false,
-											() => {
-												setTimeout(
-													() =>
-														window.location.reload(),
-													1250
-												);
-											}
-										);
-									}
-									else {
-										showNotification(
-											Liferay.Language.get(
-												'an-unexpected-error-occurred'
-											),
-											true
-										);
-									}
-								});
+	if (
+		timelineItem.status.code === WORKFLOW_STATUS_DRAFT &&
+		!!timelineItem.actions.update &&
+		discardURL
+	) {
+		dropdownItems.push({
+			href: discardURL,
+			label: Liferay.Language.get('discard'),
+			symbolLeft: 'times-circle',
+		});
+	}
+
+	if (dropdownItems && !!dropdownItems.length) {
+		return (
+			<ul className="list-unstyled" role="menu">
+				{dropdownItems.map((dropdownItem) => (
+					<li key={dropdownItem.label} role="presentation">
+						<ClayButton
+							aria-label={dropdownItem.label}
+							borderless
+							className="dropdown-item"
+							displayType="unstyled"
+							onClick={() =>
+								navigate(dropdownItem.href, dropdownItem.action)
 							}
-						},
-					});
-				},
-				symbolLeft: 'times-circle',
-			}
+						>
+							<span className="inline-item inline-item-before">
+								<ClayIcon
+									spritemap={spritemap}
+									symbol={dropdownItem.symbolLeft}
+								/>
+							</span>
+
+							{dropdownItem.label}
+						</ClayButton>
+					</li>
+				))}
+			</ul>
 		);
 	}
 
-	return (
-		<ClayDropDownWithItems
-			alignmentPosition={Align.BottomLeft}
-			items={dropdownItems}
-			spritemap={spritemap}
-			trigger={
-				<ClayButtonWithIcon
-					displayType="unstyled"
-					small
-					spritemap={spritemap}
-					symbol="ellipsis-v"
-				/>
-			}
-		/>
-	);
+	return <div>{Liferay.Language.get('no-publications-were-found')}</div>;
```

Comment:
Since this component is the dropdown menu of actions for a CTEntry instead of CTCollection, do we need this?  

Wouldn't at least some of these actions always be available for a given CTEntry? If that isn't the case, I think we should at least use a different message that more accurately describes the lack of actions for the entity.

Reply:
Oh, I forgot to remove that.
I have a check in PublicationDropdown.js that the dropdown menu won't appear at all if the user doesn't have at least "VIEW" permissions. 
At that point I wouldn't need this check either `if (dropdownItems && !!dropdownItems.length) `